### DISCUSSION TITLE
socketio client simply quits instead of reconnecting when kicked from the server

### DISCUSSION THREAD

**Discussion by dblitt** [miguelgrinberg/python-socketio#874] [Q&A]
I am trying to make my client as resilient as possible to disconnects, and it seems I've covered all the accidental disconnects. However, when the server disconnects the socket corresponding to this particular client, it fires the 'disconnect' event and simply exits the program. I would handle the reconnect myself in the disconnect, but I have no way of knowing if the client disconnected unexpectedly and the library will handle reconnecting, or if the client was kicked and the library has no intention of reconnecting.

Here is the python output when it was disconnected from the socketio server:

```
Attempting polling connection to http://socket-server:3080/socket.io/?transport=polling&EIO=4
Polling connection accepted with {'sid': 'qumAq4a_QwuBEcxVAAAQ', 'upgrades': ['websocket'], 'pingInterval': 25000, 'pingTimeout': 20000}
Engine.IO connection established
Sending packet MESSAGE data 0
Attempting WebSocket upgrade to ws://socket-server:3080/socket.io/?transport=websocket&EIO=4
WebSocket upgrade was successful
Received packet MESSAGE data 0{"sid":"0birCGRP29RUh_27AAAR"}
Namespace / is connected
Socket established to http://socket-server:3080 # my debug
Received packet MESSAGE data 1
sio unexpectedly disconnected # more debug by me, printed in @sio.on('disconnect')
Sending packet CLOSE data None
Engine.IO connection dropped
Waiting for write loop task to end
Exiting write loop task
```

And for reference, here is the node.js code powering the server that is disconnecting me:

```javascript
io.on('connection', async socket => {
    socket.disconnect();
});
```

I believe this is the expected behavior from the library. What modifications can I make to reconnect when kicked from the server?

**Comment by miguelgrinberg** [comment]
The client will not attempt a reconnection if the server issues an explicit disconnect. The reconnection support is intended for interruptions.

One option that you have is to disable the reconnections completely and handle all disconnections yourself. Another option is to use the `sio.wait()` function in your main thread to watch for a disconnection. When this function returns, it means that the client was disconnected and there is no reconnection attempt.
